arangod/Cluster/Maintenance.cpp

```diff
@@ -1536,6 +1536,59 @@ static void writeUpdateReplicatedLogLeader(
   }
 }
 
+static auto reportCurrentReplicatedLogSafeRebootIds(
+    replication2::replicated_log::QuickLogStatus const& status,
+    std::optional<replication2::agency::LogCurrent> const& current)
+    -> std::optional<
+        std::unordered_map<replication2::ParticipantId, RebootId>> {
+  auto statusSafeRebootIds = status.safeRebootIds != nullptr
+                                 ? std::optional{*status.safeRebootIds}
+                                 : std::nullopt;
+  if (!current.has_value() or current->safeRebootIds != statusSafeRebootIds) {
+    // Note that this may still be std::nullopt, which means no update.
+    return statusSafeRebootIds;
+  }
+  return std::nullopt;
+}
+
+static void writeUpdateReplicatedLogSafeRebootIds(
+    VPackBuilder& report, replication2::LogId id, DatabaseID const& dbName,
+    replication2::LogTerm& localTerm,
+    std::unordered_map<replication2::ParticipantId, RebootId> const&
+        safeRebootIds) {
+  // update Current/ReplicatedLogs/<dbname>/<logId>/safeRebootIds
+  // with precondition
+  // Plan/ReplicatedLogs/<dbname>/<logId>/term/term == currentTerm
```

Comment:
Thanks for noticing!
```suggestion
  // Plan/ReplicatedLogs/<dbname>/<logId>/currentTerm/term == localTerm
```